### PR TITLE
Added Lua 5.3.5 recipe

### PR DESCRIPTION
**Pull request by GavinNL**
Library name and version:  **lua/5.3.5**

Copied the recipe from sqlite and then modified.

Tested on Linux Mint 19.2 GCC7 using the following
```
export CONAN_HOOK_ERROR_LEVEL=40
conan create . 5.3.5@test/test
```

CMakeLists.txt file copied with permission from https://github.com/helmesjo/conan-lua

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
Hi @GavinNL ! Thanks for your contribution, could you please sign your name on the issue #4 You need be a Beta user to continue this PR.

**Comment by GavinNL** [comment]
> Hi @GavinNL ! Thanks for your contribution, could you please sign your name on the issue #4 You need be a Beta user to continue this PR.

Done!

File: recipes/lua/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    def package(self):
+        os.makedirs(os.path.join(self.package_folder, "licenses"))
+        tools.save(os.path.join(self.package_folder, "licenses", "COPYING.txt"), license)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.includedirs.append("include/lua")
```
Comment (uilianries) [inline recipes/lua/all/conanfile.py]:
```suggestion
        self.cpp_info.includedirs.append("include/lua")
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["m"]
```

Take a look in the makefile

File: recipes/lua/all/patches/0001-Remove-architecture-check-from-CMake-package.patch [hunk 2]
```diff
@@ -0,0 +1,34 @@
+From 3a53ea1f692251c052bf0b99cbc484c4f6641a8d Mon Sep 17 00:00:00 2001
```
[the remark this answers is not in the document]
Comment (GavinNL) [reply recipes/lua/all/patches/0001-Remove-architecture-check-from-CMake-package.patch]: Errr...that shouldn't be there...I copied the whole recipe from GLM I believe

File: recipes/lua/all/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,87 @@
+cmake_minimum_required(VERSION 2.8.12)
```
[the remark this answers is not in the document]
Comment (GavinNL) [reply recipes/lua/all/CMakeLists.txt]: https://github.com/helmesjo/conan-lua

[the remark this answers is not in the document]
Comment (GavinNL) [reply recipes/lua/all/CMakeLists.txt]: Should I use the one from vcpkg instead?

[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/lua/all/CMakeLists.txt]: Yes, please. It's more maintained and contain extra features, as compiler definitions.

**Comment by uilianries** [comment]
@GavinNL I've added cpp as optional, please take a look: https://github.com/GavinNL/conan-center-index/pull/1

File: recipes/lua/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -8,5 +8,10 @@ conan_basic_setup()
 
 file(GLOB SOURCE_FILES *.cpp)
 
+
+
 add_executable(${PROJECT_NAME} ${SOURCE_FILES})
 target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
+if( ${COMPILE_AS_CPP} )
```
Comment (uilianries) [inline recipes/lua/all/test_package/CMakeLists.txt]: it should be solve by Conan automatically, please, remove

File: recipes/lua/all/test_package/conanfile.py [hunk 5]
```diff
@@ -8,6 +8,9 @@ class TestPackageConan(ConanFile):
 
     def build(self):
         cmake = CMake(self)
+        # Only for the test package, so we can choose which #include header to use
+        cmake.definitions["COMPILE_AS_CPP"] = self.options["lua"].compile_as_cpp
```
Comment (uilianries) [inline recipes/lua/all/test_package/conanfile.py]: ditto

File: recipes/lua/all/CMakeLists.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (danimtb) [reply recipes/lua/all/CMakeLists.txt]: @GavinNL ping

**Comment by uilianries** [comment]
@SSE4 Will be really happy reviewing this PR, he loves Lua! :speak_no_evil:

**Pull request merged**